python/ray/train/mosaic/_mosaic_utils.py

```diff
@@ -0,0 +1,166 @@
+            return
+        self.data.update(data.items())
+        session.report(self.data)
+
+
+class RayTrainReportCallback(CheckpointSaver):
+    """A callback that wraps Composer's ``CheckpointSaver``.
+
+    This class is used to wrap each Composer ``CheckpointSaver`` to be used in Composer
+    trainer. The main role of this callback is to report the paths of the checkpoints
+    saved by the Composer ``CheckpointSaver`` it wraps. In addition, when the training
+    ends, (either with or without exception) the last checkpoint and the list of all
+    the checkpoints that have been saved and the list of Composer loggers are reported.
+
+    Example:
+        .. code-block:: python
+            # create a MosaicTrainer
+            mosaic_trainer =  MosaicTrainer(
+                    trainer_init_per_worker=trainer_init_per_worker,
+                    datasets={"train": train_dataset},
+                    trainer_init_config=trainer_init_config,
+                    scaling_config=scaling_config
+                )
+            result = mosaic_trainer.fit()
+
+            chkpt_dict = result.checkpoint.to_dict()
+
+            loggers = chkpt_dict["loggers"]
+            last_checkpoint = chkpt_dict["last_checkpoint"]
+            all_checkpoints = chkpt_dict["all_checkpoints"]
+
+    Args:
+        loggers: The list of Composer loggers that would be used in Composer
+            trainer initialization.
+        checkpoint_saver: A Composer ``CheckpointSaver`` that the callback will wrap.
+            If this argument is provided, then the parent class is initialized with the
+            passed in ``CheckpointSaver`` object's attributes. Otherwise, the parent
+            class is initialized with args provided.
+        args: Arguments for initializing a Composer ``CheckpointSaver`` object.
+    """
+
+    def __init__(self, loggers=None, checkpoint_saver: CheckpointSaver = None, **args):
+        self.loggers = loggers
+        self.last_checkpoint = None
+        self.checkpoint_count = 0
+
+        if checkpoint_saver:
+            super(RayTrainReportCallback, self).__init__(
+                checkpoint_saver.folder,
+                checkpoint_saver.filename,
+                checkpoint_saver.artifact_name,
+                checkpoint_saver.latest_filename,
+                checkpoint_saver.latest_artifact_name,
+                checkpoint_saver.save_interval,
+                overwrite=checkpoint_saver.overwrite,
+                num_checkpoints_to_keep=checkpoint_saver.num_checkpoints_to_keep,
+                weights_only=checkpoint_saver.weights_only,
+                **args
+            )
+        else:
+            super(RayTrainReportCallback, self).__init__(**args)
+
+    def close(self, state: State, logger: Logger) -> None:
+        del logger  # unused
+        checkpoint = Checkpoint.from_dict(
+            {
+                "last_checkpoint": self.last_checkpoint,
+                "loggers": self.loggers,
```

Comment:
hmmm, why reporting this?

Reply:
ah I see InMemoryLogger is used down below.
Should we only report with InMemoryLogger then?
In other words, is it guaranteed that the other loggers are serializable?

Reply:
It seems to be unnecessary to include loggers other than InMemoryLogger, so now we only report InMemoryLoggers